vcap_utils/vcap_utils/backends/backend_rpc_process.py

```diff
@@ -186,8 +198,79 @@ def _rpc_call(self, function: str, *args, **kwargs) -> Any:
         self._outgoing.put(request)
         return future.result()
 
-    def process_frame(self, *args, **kwargs) -> DETECTION_NODE_TYPE:
-        return self._rpc_call("process_frame", *args, **kwargs)
+    def process_frame(self,
+                      frame: np.ndarray,
+                      detection_node: DETECTION_NODE_TYPE,
+                      options: Dict[str, OPTION_TYPE],
+                      state: BaseStreamState) \
+            -> DETECTION_NODE_TYPE:
+        """Sends a frame to be processed by a worker process.
+        The detection_node input is carefully kept track of, to see if the
+        worker process made any modifications to it.
+        """
+        # Assign individual detection nodes with a _object_id in case they
+        # are modified in the other process, so that they can then be updated
+        input_nodes_by_id: Dict[int, DetectionNode] = {}
+        for node in self._flatten_node(detection_node):
+            node._object_id = id(node)
```

Comment:
Is there any change we could have issues with memory address reuse?

Reply:
Good question- no, because I'm holding references to these objects in the scope of the function.